### ISSUE TITLE
bagged numpy array in kwargs raises value error 

### ISSUE THREAD

**Issue by pscholle** [dask/dask#5706]
Hey everyone, I ran into this issue which only seems to occur if I provide a delayed function with keyword arguments, where I bag a numpy array. Below is a copy paste example:

```python
import dask
import dask.bag as db
import numpy as np
import dask.array as da


@dask.delayed
def status(**kwargs):
    print(kwargs)
    return


@dask.delayed
def create_array():
    return {'arr': np.array([1, 2, 3]),
            'arr1': 2}


def main():

    arrays = create_array()

    inputs = {}
    inputs['inds'] = db.from_delayed(arrays['arr'])
    inputs['sample'] = arrays['arr1']

    b = db.map(status, **inputs).compute()

    dask.delayed(b).compute()

    return


if __name__ == '__main__':
    main()
```

error message(not the traceback):
```
core.py", line 1839, in reify
    if seq and isinstance(seq[0], Iterator):
ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
```

So going into the source code and changing the function **reify()** in **core.py** from :

```python
def reify(seq):
    if isinstance(seq, Iterator):
        seq = list(seq)
    if seq and isinstance(seq[0], Iterator):
        seq = list(map(list, seq))
    return seq
``` 
to the following:
```python
def reify(seq):
    if isinstance(seq, Iterator):
        seq = list(seq)
    
     if isinstance(seq,np.ndarray):
         if seq.all() and isinstance(seq[0], Iterator):
             seq = list(map(list, seq))
    
    elif seq and isinstance(seq[0], Iterator):
        seq = list(map(list, seq))
    return seq
```
obviously solved the issue. I dont have the time to further dig around to try and understand if ther are any repercussions, but at least it works for now.

Cheers,
P

**Comment by TomAugspurger** [comment]
Thanks for the report @pscholle. A bit simpler reproducer:

```python
In [48]: obj = da.ones(10, chunks=5).to_delayed()[0]

In [49]: bag = db.from_delayed(obj)

In [50]: bag.compute()
```

I'm not sure what's best here, but I would start by seeing if the `if seq` could be replaced with `if len(seq)` without breaking any tests. Are you interested in looking into that?

**Comment by pscholle** [comment]
Sure I can give it a go. Im flying home tomorrow evening and will most likely only have time to dig into it next week

**Comment by suryaavala** [comment]
I was looking for a good first issue to tackle and this seems to be straightforward. I hope you don't mind it @pscholle

Sent #5828 your way @TomAugspurger  :)

**Comment by pscholle** [comment]
hey man @suryaavala ofcourse!! So sorry, I kind of went offline for a whole month. didnt code once.

**Comment by jrbourbeau** [comment]
Closed by #5828
